docs/hacking.md

```diff
@@ -176,3 +176,150 @@ files that serve different purposes in the compiler:
 - `RenderDocs.hs` -- Functionality for generating documentation from annotated
   carp Code.
 
+## Mini HowTos
+
+Select compiler changes are more frequent than others and have common
+high-level steps. The following sections provide some guidance on making such
+changes.
+
+### Adding a new Primitive
+
+If it doesn't require anything fancy or out of the ordinary, adding a new
+primitive to the compiler entails the following:
+
+1. Define your new primitive in `Primitives.hs`
+2. Add your primitive to the staring environment using `makePrim` in
+  `StartingEnv.hs`
+
+#### 1. Define your Primitive
+
+Primitives are functions of the `Primitive` type:
+
+```
+type Primitive = XObj -> Context -> [XObj] -> IO (Context, Either EvalError XObj)
+```
+
+Every primitive takes an xobj, the form that represents the primitive, a
+compiler context, and a list of XObjs the primitive form's arguments.
+Primitives return either a new `Context` updated based on the logic they
```

Comment:
Clarification: Primitives always return a context, and either an `XObj` or an `EvalError`.

Reply:
Ah good catch! Thank you!